app/src/main/java/io/apicurio/registry/storage/impl/jdbc/JdbcRegistryStorage.java

```diff
@@ -0,0 +1,646 @@
+        try {
+            requireNonNull(artifactId);
+            requireNonNull(ruleName);
+            requireNonNull(config);
+
+            if (getArtifactRules(artifactId).contains(ruleName))
+                throw new RuleAlreadyExistsException(ruleName);
+
+            Rule rule = Rule.builder()
+                    .artifactId(artifactId)
+                    .name(ruleName)
+                    .build();
+
+            entityManager.persist(rule);
+
+            new RuleConfigMapperUpdater()
+                    .update(config)
+                    .persistUpdate(entityManager, rule);
+
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void deleteArtifactRules(String artifactId) throws ArtifactNotFoundException, RegistryStorageException {
+        try {
+            requireNonNull(artifactId);
+
+            if (!_artifactExists(artifactId))
+                throw new ArtifactNotFoundException(artifactId);
+
+            entityManager.createQuery("DELETE FROM Rule r " +
+                    "WHERE r.artifactId = :artifact_id")
+                    .setParameter("artifact_id", artifactId)
+                    .executeUpdate();
+
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public RuleConfigurationDto getArtifactRule(String artifactId, String ruleName) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException {
+        try {
+            requireNonNull(artifactId);
+            requireNonNull(ruleName);
+
+            Rule res1 = _getRule(artifactId, ruleName);
+
+            return new RuleConfigMapperUpdater(_getRuleConfig(res1))
+                    .toRuleConfigurationDto();
+
+        } catch (EntityNotFoundException ex) {
+            throw new RuleNotFoundException(ruleName, ex);
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void updateArtifactRule(String artifactId, String ruleName, RuleConfigurationDto ruleConfiguration) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException {
+        try {
+            Rule rule = _getRule(artifactId, ruleName);
+
+            new RuleConfigMapperUpdater(_getRuleConfig(rule))
+                    .update(ruleConfiguration)
+                    .persistUpdate(entityManager, rule);
+
+        } catch (PersistenceException ex) {
+            throw new RegistryStorageException(ex);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void deleteArtifactRule(String artifactId, String ruleName) throws ArtifactNotFoundException, RuleNotFoundException, RegistryStorageException {
+        try {
+
+            if (!_artifactExists(artifactId))
```

Comment:
Perhaps make this 2 line one method -- so you don't have to copy/paste it everywhere ...

Reply:
Done, good idea